### ISSUE TITLE
Linux wrong timing in onAnimate

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#686]
Probably an initialization problem that would only happen in Linux?

**Comment by p0nce** [comment]
No reason for that to be Linux specific.

**Comment by p0nce** [comment]
This can be reproduced in Logic + x86_64 + -b release-nobounds --combined + being attached + 10.14
Probably timing dependent

**Comment by p0nce** [comment]
Several(?) reports of this being systematic on Linux

**Comment by p0nce** [comment]
Linux delta time computation is completely wrong.

**Comment by p0nce** [comment]
This is the same as #485 , not sure why is has been hidden for so long as a bug.
Time was flying 1000x too fast in terms of `dt` and `time` in `onAnimate`.

**Comment by p0nce** [comment]
Fixed in but we probably ought to do like SDL and use the monotic clock with https://linux.die.net/man/3/clock_gettime eventually.
This is really strange it hasn't been seen all the time in 3 years

**Comment by p0nce** [comment]
What do game libraries do?

- raylib for desktop => calls the GLFW one
- GLFW =>
```c
void initTime()
 {
    _glfw.timer.posix.clock = CLOCK_REALTIME;
    _glfw.timer.posix.frequency = 1000000000;

#if defined(_POSIX_MONOTONIC_CLOCK)
    struct timespec ts;
    if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
        _glfw.timer.posix.clock = CLOCK_MONOTONIC;
#endif
}

uint64_t _glfwPlatformGetTimerValue(void)
{
    struct timespec ts;
    clock_gettime(_glfw.timer.posix.clock, &ts);
    return (uint64_t) ts.tv_sec * _glfw.timer.posix.frequency + (uint64_t) ts.tv_nsec;
}
```
so CLOCK_MONOTONIC, with fallback to CLOCK_REALTIME

- SDL => `CLOCK_MONOTONIC` with fallback of `gettimeofday`

**Comment by p0nce** [comment]
Now using CLOCK_MONOTONIC preferably, with fallback of former gettimeofday, for X11 window.

**Comment by p0nce** [comment]
Fixed in v12.7.16
